h/search/logging_filters.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+"""Logging Filters."""
+
+import logging
+
+
+class FilterExceptions(logging.Filter):
+    """Filter out the specified exceptions with specified logging level."""
+
+    def __init__(self, ignore_exceptions):
+        """
+        Configure filtering out of the specified exceptions with specified logging level.
+
+        ignore_exceptions: a tuple of tuples ((exception type, loglevel))
+                           example: (("requests.exceptions.ReadTimeout", "WARNING"),)
+        """
+        logging_level_names = {level_name:
+                               (getattr(logging, level_name)
+                                if isinstance(level_name, str)
+                                else level_name)
+                               for level_name in logging._levelNames}  #pylint: disable=protected-access
```

Comment:
It is a good idea to validate the logging level argument. However I think we should avoid accessing private vars of third party packages unless really necessary. `logging._levelNames` [doesn't exist in Python 3](https://github.com/python/cpython/blob/master/Lib/logging/__init__.py) for example.

Since the list of logging levels is pretty small and doesn't change often, we could simply hardcode a list of the most common ones that we are likely to use.

Reply:
Ah ok. Didn't consider python3...I'll change it to hard code them. Unfortunately there doesn't appear to be a good way to get a list of them from the logger without accessing private vars.